app/models/manageiq/providers/hawkular/middleware_manager/alert_profile_manager.rb

```diff
@@ -7,40 +7,40 @@ def initialize(ems)
       @alerts_client = ems.alerts_client
     end
 
-    def process_alert_profile(operation, miq_alert_profile)
+    def process_alert_profile(alert_profile, miq_alert_profile)
       profile_id = miq_alert_profile[:id]
       old_alerts_ids = miq_alert_profile[:old_alerts_ids]
       new_alerts_ids = miq_alert_profile[:new_alerts_ids]
       old_assignments_ids = miq_alert_profile[:old_assignments_ids]
       new_assignments_ids = miq_alert_profile[:new_assignments_ids]
-      case operation
+      case alert_profile[:operation]
       when :update_alerts
-        update_alerts(profile_id, old_alerts_ids, new_alerts_ids, old_assignments_ids)
+        update_alerts(alert_profile[:new_assignments]["assign_to"], profile_id, old_alerts_ids, new_alerts_ids, old_assignments_ids)
       when :update_assignments
-        update_assignments(profile_id, old_alerts_ids, old_assignments_ids, new_assignments_ids)
+        update_assignments(alert_profile[:new_assignments]["assign_to"], profile_id, old_alerts_ids, old_assignments_ids, new_assignments_ids)
```

Comment:
This line and the L#18, it would only change the accessing of the resource_type:
```ruby
alert_profile[:new_assignments][:resource_type]
```
or

```ruby
alert_profile[:resource_type]
```